File: CodeWars-Solutions/Divisible-Integers.java
//Java
import java.util.ArrayList;
public class Kata
{
  public static int getCount(int n)
  {
    // your code
    int count = 0;
    String strN = Integer.toString(n);
    ArrayList<String> sNList = new ArrayList<String>();
    int[] sConvert;
    
    for (int i = 0; i < strN.length()+1; i++){
      for (int j = i; j < strN.length()+1; j++){
        sNList.add(strN.substring(i, j));
      }
    }
    for (int i = 0; i < sNList.size(); i++){
      if (sNList.get(i).equals("")){
        sNList.remove(i);
        i--;
      }
      else if (sNList.get(i).equals("0")){
        sNList.remove(i);
        i--;
      }
      else if (sNList.get(i).equals(strN)){
        sNList.remove(i);
        i--;
      }
    }
    sConvert = new int[sNList.size()];
    for (int i = 0; i < sNList.size(); i++){
      sConvert[i] = Integer.parseInt(sNList.get(i));
    }
    
    for (int i = 0; i < sConvert.length; i++){
      if (sConvert[i] != 0){
        if (n % sConvert[i] == 0){
          count++;
        }
      }
    }
    return count;
  }
}

File: CodeWars-Solutions/Divisible-Integers.py
# Python
def get_count(n):
    # your code
    count = 0
    strN = str(n)
    subIntList = list()
    subIntConvert = list()
        
# Variables
    for x in range (0, len(strN)):
        for y in range (x, len(strN)+1):
            subIntList.append(strN[x:y])
# Should fill list with every possible sub-integer
# Let's test it | First test seems to show that its missing "30", also giving weird blanks
    y = 0
    for x in range (0, len(subIntList)):
        if (subIntList[y] == ""):
            subIntList.pop(y)
            y = y -1;
        y = y + 1;
    for x in range (0, len(subIntList)):
        subIntConvert.append(int(subIntList[x]))

    for x in range (0, len(subIntConvert)):
        if (subIntConvert[x] != 0 and subIntConvert[x] != n):
            if (n % subIntConvert[x] == 0):
                count = count + 1
    return count
